docs/hosting.md

```diff
@@ -166,3 +166,57 @@ Using `docs.example.com` is an example of a subdomain. They're a simple way of c
 3. Go to the [Vercel Dashboard](https://vercel.com/dashboard) and select your Quartz project.
 4. Go to the Settings tab and then click Domains in the sidebar
 5. Enter your subdomain into the field and press Add
+
+## GitLab Pages
+
+You can configure GitLab CI to build and deploy a Quartz 4 project.
+
+In your local Quartz, create a new file `.gitlab-ci.yaml`.
+
+```yaml title="quartz/.gitlab-ci.yaml"
+stages:
+  - build_and_test
+  - deploy
+
+variables:
+  NODE_VERSION: '18.14'
+
+build_and_test:
+  stage: build_and_test
+  rules:
+    - if: '$CI_COMMIT_REF_NAME == "v4"'
+  before_script:
+    - apt-get update -q && apt-get install -y nodejs npm
+    - npm install -g n
+    - n $NODE_VERSION
+    - hash -r
+    - npm ci
+  script:
+    - npx prettier --write .
+    - npm run check
+    - npm test
```

Comment:
`npm test` doesn't need to be run for most people :) just for maintainers usually

Reply:
I see, might as well remove it to match the GitHub instructions on the page!